### PR TITLE
[Android] Fix onNOCChainGeneration terminate issue

### PR DESCRIPTION
**Pull request by joonhaengHeo**
When onNOCChainGeneration is called, if a NOC is not issued or appropriate parameters are not set for other reasons, commissioning does not fail and no action is taken.

To fix this, when an error occurs, a null value is set to enable subsequent operation, so that commissioning fails and ends.

### CODE REVIEW

File: src/controller/java/CHIPDeviceController-JNI.cpp [hunk 1]
```diff
@@ -171,44 +171,54 @@ JNI_METHOD(jint, onNOCChainGeneration)
...
+    jmethodID getIpk;
+    jmethodID getAdminSubject;
+
+    jbyteArray rootCertificate         = nullptr;
+    jbyteArray intermediateCertificate = nullptr;
+    jbyteArray operationalCertificate  = nullptr;
+    jbyteArray ipk                     = nullptr;
+
+    Optional<NodeId> adminSubjectOptional;
+    uint64_t adminSubject;
```
Comment (yunhanw-google) [inline src/controller/java/CHIPDeviceController-JNI.cpp:185]: init this value

File: src/controller/java/CHIPDeviceController-JNI.cpp [hunk 2]
```diff
@@ -171,44 +171,54 @@ JNI_METHOD(jint, onNOCChainGeneration)
...
     err = chip::JniReferences::GetInstance().FindMethod(env, controllerParams, "getAdminSubject", "()J", &getAdminSubject);
-    VerifyOrReturnValue(err == CHIP_NO_ERROR, static_cast<jint>(err.AsInteger()));
-
-    jbyteArray rootCertificate = (jbyteArray) env->CallObjectMethod(controllerParams, getRootCertificate);
-    VerifyOrReturnValue(rootCertificate != nullptr, static_cast<jint>(CHIP_ERROR_BAD_REQUEST.AsInteger()));
+    VerifyOrExit(err == CHIP_NO_ERROR, ChipLogError(Controller, "Find getAdminSubject method fail!"));
 
-    jbyteArray intermediateCertificate = (jbyteArray) env->CallObjectMethod(controllerParams, getIntermediateCertificate);
-    VerifyOrReturnValue(intermediateCertificate != nullptr, static_cast<jint>(CHIP_ERROR_BAD_REQUEST.AsInteger()));
+    rootCertificate = (jbyteArray) env->CallObjectMethod(controllerParams, getRootCertificate);
```
Comment (yunhanw-google) [inline src/controller/java/CHIPDeviceController-JNI.cpp:212]: please use static_cast<jbyteArray>

File: src/controller/java/CHIPDeviceController-JNI.cpp [hunk 3]
```diff
@@ -171,44 +171,54 @@ JNI_METHOD(jint, onNOCChainGeneration)
...
+    VerifyOrExit(rootCertificate != nullptr, err = CHIP_ERROR_BAD_REQUEST);
 
-    jbyteArray operationalCertificate = (jbyteArray) env->CallObjectMethod(controllerParams, getOperationalCertificate);
-    VerifyOrReturnValue(operationalCertificate != nullptr, static_cast<jint>(CHIP_ERROR_BAD_REQUEST.AsInteger()));
+    intermediateCertificate = (jbyteArray) env->CallObjectMethod(controllerParams, getIntermediateCertificate);
```
Comment (yunhanw-google) [inline src/controller/java/CHIPDeviceController-JNI.cpp:215]: please use static_cast

File: src/controller/java/CHIPDeviceController-JNI.cpp (hunk 1 above)
Comment (joonhaengHeo) [reply src/controller/java/CHIPDeviceController-JNI.cpp:185]: added it.

File: src/controller/java/CHIPDeviceController-JNI.cpp (hunk 3 above)
Comment (joonhaengHeo) [reply src/controller/java/CHIPDeviceController-JNI.cpp]: modified.

File: src/controller/java/CHIPDeviceController-JNI.cpp (hunk 2 above)
Comment (joonhaengHeo) [reply src/controller/java/CHIPDeviceController-JNI.cpp]: modified.

**Review by yunhanw-google — APPROVED** [review]

**Review by sharadb-amazon — APPROVED** [review]

**Pull request merged**